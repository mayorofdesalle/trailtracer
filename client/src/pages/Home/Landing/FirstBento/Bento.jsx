import { useTheme } from 'styled-components';

import useWindowSize from '@hooks/useWindowSize';

import GridSizeProvider from '../GridSizeProvider';
import { ActionBoxFirst, ActionBoxSecond } from './ActionBox';
import BentoGrid from './BentoGrid';
import HeroDescriptionBox from './HeroDescriptionBox';
import HeroImageBox from './HeroImageBox';
import NewsBox from './NewsBox';
import NewsImageBox from './NewsImageBox';
import PopularBox from './PopularBox';
import TopTrailBox from './TopTrailBox';

/**
 * Bento
 * @description
 * First bento of the landing page.
 **/
const Bento = () => {
    const theme = useTheme();
    const { height, width } = useWindowSize();
    const largeScreen = width > theme.breakpoints.medium && width / height > 1;

    return (
        <BentoGrid>
            <HeroImageBox />
            <HeroDescriptionBox />
            <ActionBoxFirst />
            <ActionBoxSecond />

            {largeScreen && (
                <>
                    <NewsImageBox $color={theme.colors.primary} />
                    <NewsBox $glass />
                    <TopTrailBox $color={theme.colors.primary} />
                    <PopularBox $glass />
                </>
            )}

            <GridSizeProvider gridArea='Slider' />
            <GridSizeProvider gridArea='StartButton' />
        </BentoGrid>
    );
};

export default Bento;
