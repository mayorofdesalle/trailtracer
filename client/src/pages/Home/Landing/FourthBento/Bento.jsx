import styled, { useTheme } from 'styled-components';

import { BentoBox } from '@components/ui/bento';

import GridSizeProvider from '../GridSizeProvider';
import BentoGrid from './BentoGrid';


const Box1 = styled(BentoBox)`
    grid-area: Box1;
`;

const Box2 = styled(BentoBox)`
    grid-area: Box2;
`;

const Box3 = styled(BentoBox)`
    grid-area: Box3;
`;

const Box4 = styled(BentoBox)`
    grid-area: Box4;
`;

const Box5 = styled(BentoBox)`
    grid-area: Box5;
`;

/**
 * Bento
 * @description
 * Fourth bento of the landing page.
 **/
const Bento = () => {
    const theme = useTheme();

    return (
        <BentoGrid >
            <Box1 $color={theme.colors.primary} />
            <Box2 $color={theme.colors.secondary} />
            <Box3 $glass />
            <Box4 $glass />
            <Box5 $glass />
            <GridSizeProvider gridArea='Slider' />
            <GridSizeProvider gridArea='StartButton' />
        </BentoGrid>
    );
};

export default Bento;
