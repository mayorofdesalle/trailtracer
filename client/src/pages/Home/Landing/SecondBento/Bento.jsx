import styled, { useTheme } from 'styled-components';

import BentoBox from '@components/ui/Bento/BentoBox';

import GridSizeProvider from '../GridSizeProvider';
import BentoGrid from './BentoGrid';

const Box1 = styled(BentoBox)`
    grid-area: Box1;
`;

const Box2 = styled(BentoBox)`
    grid-area: Box2;
`;

const Box3 = styled(BentoBox)`
    grid-area: Box3;
`;

const Box4 = styled(BentoBox)`
    grid-area: Box4;
`;

const Box5 = styled(BentoBox)`
    grid-area: Box5;
`;

/**
 * Bento
 * @description
 * Second bento of the landing page.
 **/
const Bento = () => {
    const theme = useTheme();

    return (
        <BentoGrid >
            <Box1 $color={theme.colors.primary} />
            <Box2 $glass />
            <Box3 $color={theme.colors.primary} />
            <Box4 $glass />
            <Box5 $color={theme.colors.secondary} />
            <GridSizeProvider gridArea='Slider' />
            <GridSizeProvider gridArea='StartButton' />
        </BentoGrid>
    );
};

export default Bento;
