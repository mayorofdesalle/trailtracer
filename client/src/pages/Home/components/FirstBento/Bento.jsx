import { useTheme } from 'styled-components';
import PropTypes from 'prop-types';

import bikerImg from '@assets/images/biker.svg';
import arrowLeftImg from '@assets/images/arrowLeft.svg';
import arrowUpRightImg from '@assets/images/arrowUpRight.svg';
import arrowUpImg from '@assets/images/arrowUp.svg';
import useWindowSize from '@hooks/useWindowSize';

import Text, { Prim, Sec } from '@components/ui/Text';
import Vector from '@components/ui/Vector';
import { convertRemToPixels } from '@utils/screenMath';

import BentoGridContainer from './BentoGridContainer';
import HeroImageBox from './HeroImageBox';
import HeroDescriptionBox from './HeroDescriptionBox';
import ActionBox from './ActionBox';
import NewsImageBox from './NewsImageBox';
import NewsBox from './NewsBox';
import TopTrailBox from './TopTrailBox';
import PopularBox from './PopularBox';

/**
 * Bento
 * @description
 * This component represents the main buildign block of the Home page.
 * It displays a grid layout with various sections including a hero image, description, action boxes, slider, and buttons.
 * The component is responsive and adjusts its layout based on the window size.
 **/
const Bento = ({ children }) => {
    const theme = useTheme();
    const mediumBreakpoint = convertRemToPixels(theme.breakpoints.large.slice(0, -3));
    const { width } = useWindowSize();

    return (
        <BentoGridContainer>
            <HeroImageBox $gridArea='HeroImageBox'>
                <img src={bikerImg} alt='A biker on an enduro motorcycle' />
            </HeroImageBox>

            <HeroDescriptionBox $gridArea='HeroDescriptionBox'>
                <Text $heading $color={theme.colors.text}>
                    <Sec>COLLABORATE</Sec> WITH RIDERS LIKE YOU <Sec>AND PLAN</Sec> <Prim>YOUR NEXT TRAIL!</Prim>
                </Text>
            </HeroDescriptionBox>

            <ActionBox $gridArea='ActionBox1' $color={theme.colors.primary}>
                <Vector id='L' $image={width > mediumBreakpoint ? arrowUpRightImg : arrowUpImg} $color={theme.colors.background} $height='7svh' $width='7svh' $position='top' />
                <Text>NAVIGATE TO LEARN MORE</Text>
            </ActionBox>

            <ActionBox $gridArea='ActionBox2' $color={theme.colors.secondary} >
                <Text $color={theme.colors.primary}>OR START YOUR JOURNEY NOW</Text>
                <Vector id='R' $image={arrowLeftImg} $color={theme.colors.primary} $height={'4svh'} $width={'8svh'} $position='bottom' />
            </ActionBox>

            {width > mediumBreakpoint && (
                <>
                    <NewsImageBox $gridArea='NewsImageBox' />
                    <NewsBox $gridArea='NewsBox' />
                    <PopularBox $gridArea='PopularBox' />
                    <TopTrailBox $gridArea='TopTrailBox' />
                </>
            )}

            {children}

        </BentoGridContainer>
    );
};

Bento.propTypes = {
    children: PropTypes.node
};

export default Bento;
