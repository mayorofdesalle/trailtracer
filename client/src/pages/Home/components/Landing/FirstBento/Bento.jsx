import { useTheme } from 'styled-components';

import bikerImg from '@assets/images/biker.svg';
import useWindowSize from '@hooks/useWindowSize';
import Text, { Prim, Sec } from '@components/ui/Text';
import Vector from '@components/ui/Vector';
import { remToPx } from '@utils/conversions';

import GridPlaceholder from '../GridPlaceholder';
import BentoGrid from './BentoGrid';
import HeroImageBox from './HeroImageBox';
import HeroDescriptionBox from './HeroDescriptionBox';
import ActionBox from './ActionBox';
import NewsImageBox from './NewsImageBox';
import NewsBox from './NewsBox';
import TopTrailBox from './TopTrailBox';
import PopularBox from './PopularBox';

/**
 * Bento
 * @description
 * This is the first bento of the landing page.
 **/
const Bento = () => {
    const theme = useTheme();
    const { height, width } = useWindowSize();
    const largeBreakpoint = remToPx(theme.breakpoints.large.slice(0, -3));
    const aspectRatio = width / height;

    return (
        <BentoGrid>
            <HeroImageBox $color={theme.colors.secondary}>
                <img src={bikerImg} alt='A biker on an enduro motorcycle' />
            </HeroImageBox>

            <HeroDescriptionBox $glass>
                <Text $heading $color={theme.colors.text}>
                    <Sec>COLLABORATE</Sec> WITH RIDERS LIKE YOU <Sec>AND PLAN</Sec> <Prim>YOUR NEXT TRAIL!</Prim>
                </Text>
            </HeroDescriptionBox>

            <ActionBox $gridArea='ActionBox1' $color={theme.colors.primary}>
                <Vector id='L' $name={width > largeBreakpoint ? 'arrowUpRight' : 'arrowUp'} $color={theme.colors.background} $height='7svh' $width='7svh' $position='top' />
                <Text>NAVIGATE TO LEARN MORE</Text>
            </ActionBox>

            <ActionBox $gridArea='ActionBox2' $color={theme.colors.secondary} >
                <Text $color={theme.colors.primary}>OR START YOUR JOURNEY NOW</Text>
                <Vector id='R' $name='arrowLeft' $color={theme.colors.primary} $height={'4svh'} $width={'8svh'} $position='bottom' />
            </ActionBox>

            {width > largeBreakpoint && aspectRatio > 1 && (
                <>
                    <NewsImageBox $color={theme.colors.primary} />
                    <NewsBox $glass />
                    <TopTrailBox $color={theme.colors.primary} />
                    <PopularBox $glass />
                </>
            )}

            <GridPlaceholder gridArea='Slider' />
            <GridPlaceholder gridArea='StartButton' />
        </BentoGrid>
    );
};

export default Bento;
