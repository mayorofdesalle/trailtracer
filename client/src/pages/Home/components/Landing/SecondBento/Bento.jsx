import styled from 'styled-components';

import BentoBox from '@components/ui/BentoBox';
import { glassMorph } from '@components/misc/mixins';

import BentoGridContainer from './BentoGridContainer';


const Box1 = styled(BentoBox)`
    background-color: ${({ theme }) => theme.colors.primary};
`;

const Box2 = styled(BentoBox)`
    ${glassMorph};
    border: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
`;

const Box3 = styled(BentoBox)`
    background-color: ${({ theme }) => theme.colors.primary};
`;

const Box4 = styled(BentoBox)`
    ${glassMorph};
    border: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
`;

const Box5 = styled(BentoBox)`
    background-color: ${({ theme }) => theme.colors.secondary};
`;

/**
 * Bento
 * @description
 * This component represents the main buildign block of the Home page.
 * It displays a grid layout with various sections including a hero image, description, action boxes, slider, and buttons.
 * The component is responsive and adjusts its layout based on the window size.
 **/
const Bento = () => {
    return (
        <BentoGridContainer >
            <Box1 $gridArea='Box1' />
            <Box2 $gridArea='Box2' />
            <Box3 $gridArea='Box3' />
            <Box4 $gridArea='Box4' />
            <Box5 $gridArea='Box5' />
        </BentoGridContainer>
    );
};

export default Bento;
