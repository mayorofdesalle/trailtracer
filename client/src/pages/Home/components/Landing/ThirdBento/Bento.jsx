import styled, { useTheme } from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';

import GridPlaceholder from '../GridPlaceholder';
import BentoGrid from './BentoGrid';

const Box1 = styled(BentoBox)`
    grid-area: Box1;
`;

const Box2 = styled(BentoBox)`
    grid-area: Box2;
`;

const Box3 = styled(BentoBox)`
    grid-area: Box3;
`;

const Box4 = styled(BentoBox)`
    grid-area: Box4;
`;

const Box5 = styled(BentoBox)`
    grid-area: Box5;
`;

/**
 * Bento
 * @description
 * This is the third bento of the landing page.
 **/
const Bento = () => {
    const theme = useTheme();

    return (
        <BentoGrid >
            <Box1 $color={theme.colors.secondary} />
            <Box2 $color={theme.colors.primary} />
            <Box3 $color={theme.colors.text} />
            <Box4 $glass />
            <Box5 $glass />
            <GridPlaceholder gridArea='Slider' />
            <GridPlaceholder gridArea='StartButton' />
        </BentoGrid>
    );
};

export default Bento;
