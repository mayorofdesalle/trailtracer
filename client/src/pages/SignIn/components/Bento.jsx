import { useTranslation } from 'react-i18next';
import { useTheme } from 'styled-components';

import Text from '@components/ui/Text';
import Icon from '@components/ui/Icon';
import Button from '@components/ui/Button';
import useWindowSize from '@hooks/useWindowSize';

import BentoGrid from './BentoGrid';
import SignupBox from './SignupBox';
import ImageBox from './ImageBox';
import SigninBox from './SigninBox/SigninBox';
import ButtonBox from './ButtonBox';
import MapBox from './MapBox';

const Bento = () => {
    const theme = useTheme();
    const { t } = useTranslation();
    const { height, width } = useWindowSize();
    const largeScreen = width > theme.breakpoints.medium && width / height > 1;

    return (
        <BentoGrid>
            <ImageBox />

            <SignupBox $glass>
                <Text $color={theme.colors.primary} $heading $ratio={largeScreen ? 1 : 0.8}>{t('signinPage.signupTitle')}</Text>
                {largeScreen && (
                    <Text $color={theme.colors.text}>
                        {t('signinPage.explanation')}
                    </Text>
                )}
                <Button $bgColor={theme.colors.secondary}>
                    <Text $color={theme.colors.text} $heading>{t('signinPage.signup')}</Text>
                </Button>
            </SignupBox>

            {largeScreen && <MapBox />}

            <SigninBox />

            <ButtonBox as={Button} $gridArea='TermsBox'>
                <Icon name='article-fill' color={theme.colors.primary} />
                <Text $color={theme.colors.text} $heading $ratio={largeScreen ? 0.7 : 0.5}>
                    {t('signinPage.terms')}
                </Text>
            </ButtonBox>

            <ButtonBox as={Button} $gridArea='HelpBox'>
                <Icon name='login-box-fill' color={theme.colors.primary} />
                <Text $color={theme.colors.text} $heading $ratio={0.8}>{t('signinPage.help')}</Text>
            </ButtonBox>
        </BentoGrid>
    );
};

export default Bento;