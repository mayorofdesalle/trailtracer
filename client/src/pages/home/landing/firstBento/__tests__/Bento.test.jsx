import { describe, expect, it, render, screen } from '@app/tester';

import Bento from '../Bento';

describe('Bento', () => {

    it('should render correctly', () => {
        render(<Bento data-testid='bento' />);
        const bento = screen.getByTestId('bento');

        expect(bento).toBeInTheDocument();
    });

});