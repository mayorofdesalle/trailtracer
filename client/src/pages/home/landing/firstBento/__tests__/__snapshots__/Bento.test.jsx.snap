// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bento > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh sc-dmyCSP sc-eTNRI jVdZA igIKKw ypUGI"
    role="grid"
  >
    <div
      class="sc-beySPh sc-guDLey sc-ktwOfi jVdZA lbUchL binSLc"
    >
      <img
        alt="A supersport biker"
        src="file:///C:/src/assets/images/biker.png"
      />
    </div>
    <div
      class="sc-beySPh sc-guDLey sc-brSamD jVdZA cVargU zyufv"
    >
      <div
        class="sc-fsYfdN fkLXpT"
      >
        landingPage.title
      </div>
    </div>
    <div
      class="sc-beySPh sc-guDLey sc-csKJxZ jVdZA hEuWzj exsIFa"
    >
      <div
        class="sc-irLvIq bXrkAM left"
        role="img"
      />
      <div
        class="sc-fsYfdN jbLRtD"
      >
        landingPage.CTA1
      </div>
    </div>
    <div
      class="sc-beySPh sc-guDLey sc-csKJxZ jVdZA gasxRO nnRwN"
    >
      <div
        class="sc-fsYfdN kzpmpg"
      >
        landingPage.CTA2
      </div>
      <div
        class="sc-irLvIq iRAYUb right"
        role="img"
      />
    </div>
    <div
      class="sc-beySPh sc-guDLey sc-fmKFGs jVdZA hEuWzj kGiOzI"
    />
    <div
      class="sc-beySPh sc-guDLey sc-hABBmJ jVdZA cVargU kiAnSg"
    />
    <div
      class="sc-beySPh sc-guDLey sc-blmEgr jVdZA hEuWzj dUzXWM"
    />
    <div
      class="sc-beySPh sc-guDLey sc-fHejqy jVdZA cVargU cNCZWH"
    />
    <div
      class="sc-beySPh sc-guDLey sc-hLQSwg jVdZA lbUchL cBPFij"
    />
    <div
      class="sc-beySPh sc-guDLey sc-hLQSwg jVdZA lbUchL fvAXYY"
    />
  </div>
</DocumentFragment>
`;
